nep-x1.mediawiki

```diff
@@ -0,0 +1,161 @@
+<pre>
+  NEP: Undefined
+  Title: NFT Royalty Standard
+  Author: Owen Zhang<zhangzhihong@ngd.neo.org>, Vincent Geneste<vincentgeneste@gmail.com>
+  Type: Standard
+  Status: Draft
+  Created: 2022-09-05
+</pre>
+
+==Abstract==
+
+This NEP defines a global standard to get royalty payment information for Non-Fungible Tokens (NFTs) in order to enable support for royalty payments across all NFT marketplaces in the NEO Smart Economy.
+
+==Motivation==
+
+There is a need for supporting royalty information, which is useful not only for royalty payments across NFT marketplaces but also good to protect creative copyright. Currently, there is no standard for creators to declare royalty to all marketplaces in their NFTs. In this sense, they have to discuss the royalty with different marketplaces, otherwise these marketplaces, which haven't touched with the creators, can't get royalty details and define the royalty in their marketplace contract by themselves. This standard is compatible with [NEP-11](https://github.com/neo-project/proposals/blob/master/nep-11.mediawiki).
+
+==Specification==
+
+NEP-11 compliant contracts MAY implement this NEP for royalties to provide a standard method of specifying royalty payment information.
+
+===Common methods===
+
+====royaltyInfo====
+
+<pre>
+{
+  "name": "royaltyInfo",
+  "safe": true,
+  "parameters": [
+    {
+      "name": "tokenId",
+      "type": "ByteString"
+    },
+    {
+      "name": "royaltyToken",
+      "type": "Hash160"
+    },
+    }
+    {
+      "name": "salePrice",
+      "type": "Integer"
+    }
+  ],
+  "returntype": "Array"
```

Comment:
Should this method always return the same result given the same input data or it's allowed to change royalty amount or receipts through time? Maybe a note should be added. Ref. https://github.com/neo-project/proposals/blob/90686c5fae075541285f1200a584223504878b6d/nep-11.mediawiki?plain=1#L40

Reply:
`royaltyAmount` maybe dynamic. `royaltyRecipient` can be tranferred to another address.